dart/math/MathTypes.h

```diff
@@ -83,8 +85,28 @@ typedef Matrix<double, 6, 6> Matrix6d;
 typedef std::vector<Eigen::Vector3d> EIGEN_V_VEC3D;
 typedef std::vector<std::vector<Eigen::Vector3d > > EIGEN_VV_VEC3D;
 
+template <typename _Tp>
+using aligned_vector = std::vector<_Tp, Eigen::aligned_allocator<_Tp>>;
+
+template <typename _Key, typename _Tp, typename _Compare = std::less<_Key>>
+using aligned_map = std::map<_Key, _Tp, _Compare,
+    Eigen::aligned_allocator<std::pair<const _Key, _Tp>>>;
+
+template<typename _Tp, typename... _Args>
+inline std::shared_ptr<_Tp> make_aligned_shared()
```

Comment:
I'm not clear on what the purpose of this overload is. Normally with variadic templates, an argument-free overload is used to terminate the template recursion, but we're not actually using template recursion in this case.

Also, this version is using regular ``std::make_shared`` instead of ``std::allocate_shared``, which seems to defeat the purpose, doesn't it?